sql/solutions/lesson2.md

```diff
@@ -0,0 +1,150 @@
+---
+layout: page
+title: Introduction to SQL Solutions
+---
+# Lesson 2 solutions to exercises
+
+## L2.1 Show the names of all the people. Sort the result by name in ascending (alphabetical) order, in case of equal name sort by country of residence in alphabetical order.
+
+```SQL
+SELECT name
+  FROM people
+ ORDER BY name ASC, country_of_residence ASC;
+```
+
+##  L2.2 Show the name and the country of birth of the person with highest number of children.
+
+```SQL
+SELECT name, country_of_birth
+  FROM people
+ ORDER BY number_of_children DESC
+ LIMIT 1;
+```
+
+##  L2.3 Show the country of birth of the oldest person.
+
+```SQL
+SELECT country_of_birth
+  FROM people
+ ORDER BY age DESC
+ LIMIT 1;
+```
+
+##  L2.4 Show the age of the oldest person.
+
+```SQL
+SELECT age
+  FROM people
+ ORDER BY age DESC
+ LIMIT 1;
+```
+
+##  L2.5 Show the location of cities in UK.
+
+Solution 1:
+```SQL
+SELECT city
+  FROM locations
+ WHERE country = 'UK';
+```
+
+Solution 2:
+```SQL
+SELECT DISTINCT city
+  FROM locations
+ WHERE country = 'UK';
+```
+
+##  L2.6 Show all the host names.
+
+Solution 1:
+```SQL
+SELECT name
+  FROM hosts;
+```
+
+Solution 2:
+```SQL
+SELECT DISTINCT name
+  FROM hosts;
+```
+
+##  L2.7 Show the host name and the student capacity. Sort the result in capacity descending order.
+
+Solution 1:
+```SQL
+SELECT name, students_capacity
+  FROM hosts
+ ORDER BY students_capacity DESC;
+```
+
+Solution 2:
+```SQL
+SELECT DISTINCT name, students_capacity
+  FROM hosts
+ ORDER BY students_capacity DESC;
+```
+
+## L2.8 Pick a host ID and show the dates where workshops were held there. Order the result chronologically.
+
+Solution 1:
+```SQL
+SELECT *
+  FROM workshops
+ WHERE host_id = 7
+ ORDER BY workshop_date;
```

Comment:
I think Solution 1 is more accurate in terms of the _chronologically_ request. But note that it only asks for the date. You may need to re-think the usage of the * in this one :)